02_Util/src/fileAccess/s3Storage.ts

```diff
@@ -2,22 +2,64 @@
 //
 // SPDX-License-Identifier: Apache 2.0
 
-import { IFileAccess } from '@citrineos/base';
+import { IFileAccess, SystemConfig } from '@citrineos/base';
+import AWS from 'aws-sdk';
 
 export class S3Storage implements IFileAccess {
-  getFile(id: string): Promise<Buffer> {
-    // TODO: implement the logic
-    throw new Error(`Get file ${id} not yet implemented.`);
+  private _s3: AWS.S3;
+  protected readonly _config: SystemConfig;
+
+  constructor(config: SystemConfig) {
+    this._config  = config;
+
+    this._s3 = new AWS.S3({
+      endpoint: `http://${this._config.util.s3Storage?.endpointHost || 'localstack'}:${this._config.util.s3Storage?.endpointPort || '4566'}`,
+      accessKeyId: this._config.util.s3Storage?.accessKeyId as string || 'null',
+      secretAccessKey: this._config.util.s3Storage?.secretAccessKey as string || 'null',
+      s3ForcePathStyle: true,
+    });
   }
 
-  uploadFile(
+  async getFile(id: string): Promise<Buffer> {
+    const bucketName: string = this._config.util.s3Storage?.bucketName as string;
+
+    try {
+      const result = await this._s3
+        .getObject({
+          Bucket: bucketName || 'citrineos-s3-bucket',
+          Key: id,
+        })
+        .promise();
+
+      if (result.Body instanceof Buffer) {
+        return result.Body;
+      } else {
+        throw new Error('File content is not in Buffer format.');
+      }
+    } catch (e) {
+      throw new Error(`Failed to get file ${id}: ${e}`);
+    }
+  }
+
+  async uploadFile(
     fileName: string,
     content: Buffer,
     filePath?: string,
   ): Promise<string> {
-    // TODO: implement the logic
-    throw new Error(
-      `Upload file ${fileName} to ${filePath} with content ${content} not yet implemented.`,
-    );
+    const bucketName: string = this._config.util.s3Storage?.bucketName as string;
+    try {
+      const result = await this._s3
+        .upload({
+          Bucket: bucketName || 'citrineos-s3-bucket',
+          Key: `${filePath}${fileName}`,
+          Body: content,
+          ContentType: 'application/octet-stream',
+        })
+        .promise();
+
+      return result.Location;
```

Comment:
This return value isn't equivalent to the fileId used to fetch the file in the getFile method. result.Location is the url to the file. In order to correctly implement the interface, this needs to return just the file id, so that:
```
iFileAccess.getFile(iFileAccess.uploadFile([...]));
```
would work.

To make this clearer in the future, we should add javadoc descriptions to these methods on the interface as part of this PR.